modules/core/perf/opencl/perf_usage_flags.cpp

```diff
@@ -12,25 +12,33 @@
 namespace opencv_test {
 namespace ocl {
 
-typedef TestBaseWithParam<tuple<cv::Size, bool> > UsageFlagsBoolFixture;
-
-OCL_PERF_TEST_P(UsageFlagsBoolFixture, UsageFlags_AllocHostMem, ::testing::Combine(OCL_TEST_SIZES, Bool()))
+typedef TestBaseWithParam<tuple<cv::Size, UMatUsageFlags, UMatUsageFlags, UMatUsageFlags>> SizeUsageFlagsFixture;
+
+OCL_PERF_TEST_P(SizeUsageFlagsFixture, UsageFlags_AllocMem,
+    ::testing::Combine(
+        OCL_TEST_SIZES,
+        testing::Values(USAGE_DEFAULT, USAGE_ALLOCATE_HOST_MEMORY, USAGE_ALLOCATE_DEVICE_MEMORY), // USAGE_ALLOCATE_SHARED_MEMORY
+        testing::Values(USAGE_DEFAULT, USAGE_ALLOCATE_HOST_MEMORY, USAGE_ALLOCATE_DEVICE_MEMORY), // USAGE_ALLOCATE_SHARED_MEMORY
+        testing::Values(USAGE_DEFAULT, USAGE_ALLOCATE_HOST_MEMORY, USAGE_ALLOCATE_DEVICE_MEMORY) // USAGE_ALLOCATE_SHARED_MEMORY
```

Comment:
These tests are very long (>50% of whole opencv_perf_core):

> `108 tests from OCL_SizeUsageFlagsFixture_UsageFlags_AllocMem (104914 ms total)`

Perhaps we should replace OCL_TEST_SIZES with `::testing::Values(sz1080p)`

and revert this change:

> `OCL_TEST_CYCLE_MULTIRUN(500)`

Reply:
As I wrote in the OP...this PR is NOT NOT NOT ready to be merged. I have been seeking code and performance review. I have those two now...so I will start looking at 3.x and if I can readily backport this.

These long perf tests will not be part of the final PR. They are only in the current changeset so that I can show you the tests that I run to generate the above performance reports, and to have broader coverage on more platforms than I have locally. They will be used in my effort to backport to 3.x (if that is readily possible).

I will definitely remove the OCL_TEST_CYCLE_MULTIRUN() in final code. AT the same time, I will evaluate if I can continue to use `OCL_TEST_SIZES`. We might even consider making this perf test disabled.

Reply:
updated PR no longer loops 500. On my test machine, the test you write above is smaller than other perf tests, e.g. the Dft test is 3 times longer.

```
144 tests from OCL_DftFixture_Dft (39435 ms total)
...
108 tests from OCL_SizeUsageFlagsFixture_UsageFlags_AllocMem (13932 ms total)
```

Is this length ok?
If you still want to remove sizes, then I recommend we keep only `OCL_SIZE_1` (which is szVGA) and `OCL_SIZE_4` (which is sz2160p) so that we have the smallest, and largest. That will probably half again the run time.

Reply:
the default CI on linux x64 opencl and windows opencl shows the perf test runs in single-digit seconds, or in less than one second. Dft is still larger on both.

I think we can keep the perf test as it is now. Agree?
If yes, then I have no further work to do and, from my eval, this PR is ready to merge into master.